include/pybind11/pybind11.h

```diff
@@ -1169,90 +1169,107 @@ class class_ : public detail::generic_type {
+                .def("__gt__", [](const Type &value, Scalar value2) { return (Scalar) value > value2; })
+                .def("__le__", [](const Type &value, Scalar value2) { return (Scalar) value <= value2; })
+                .def("__ge__", [](const Type &value, Scalar value2) { return (Scalar) value >= value2; })
+                .def("__invert__", [](const Type &value) { return ~((Scalar) value); })
+                .def("__and__", [](const Type &value, Scalar value2) { return (Scalar) value & value2; })
+                .def("__or__", [](const Type &value, Scalar value2) { return (Scalar) value | value2; })
+                .def("__xor__", [](const Type &value, Scalar value2) { return (Scalar) value ^ value2; })
+                .def("__rand__", [](const Type &value, Scalar value2) { return (Scalar) value & value2; })
+                .def("__ror__", [](const Type &value, Scalar value2) { return (Scalar) value | value2; })
+                .def("__rxor__", [](const Type &value, Scalar value2) { return (Scalar) value ^ value2; })
+                .def("__and__", [](const Type &value, const Type &value2) { return (Scalar) value & (Scalar) value2; })
+                .def("__or__", [](const Type &value, const Type &value2) { return (Scalar) value | (Scalar) value2; })
+                .def("__xor__", [](const Type &value, const Type &value2) { return (Scalar) value ^ (Scalar) value2; });
             }
         }
-        def("__hash__", [](const Type &value) { return (Scalar) value; });
+        cls
+        .def("__hash__", [](const Type &value) { return (Scalar) value; })
         // Pickling and unpickling -- needed for use with the 'multiprocessing' module
-        def("__getstate__", [](const Type &value) { return pybind11::make_tuple((Scalar) value); });
-        def("__setstate__", [](Type &p, tuple t) { new (&p) Type((Type) t[0].cast<Scalar>()); });
+        .def("__getstate__", [](const Type &value) { return pybind11::make_tuple((Scalar) value); })
+        .def("__setstate__", [](Type &p, tuple t) { new (&p) Type((Type) t[0].cast<Scalar>()); });
     }
 
     /// Export enumeration entries into the parent scope
-    enum_& export_values() {
+    enum_ export_values() && {
         for (const auto &kv : m_entries)
             m_parent.attr(kv.first) = kv.second;
-        return *this;
+        return std::move(*this);
     }
 
     /// Add an enumeration entry
-    enum_& value(char const* name, Type value) {
+    enum_ value(char const* name, Type value) && {
         auto v = pybind11::cast(value, return_value_policy::copy);
-        this->attr(name) = v;
+        cls.attr(name) = v;
         m_entries[pybind11::str(name)] = v;
-        return *this;
+        return std::move(*this);
+    }
+
+    /// Get the associated class object.
+    class_<Type> into_class() && {
+        return cls;
+    }
+
+    operator class_<Type>() && {
```

Comment:
That's how I had it initially (in the other PR) -- with hindsight, this could probably be removed.